fdbclient/ClientKnobs.cpp

```diff
@@ -190,6 +190,7 @@ void ClientKnobs::initialize(Randomize randomize) {
 	init( BACKUP_STATUS_JITTER,                   0.05 );
 	init( MIN_CLEANUP_SECONDS,                  3600.0 );
 	init( FASTRESTORE_ATOMICOP_WEIGHT,               1 ); if( randomize && BUGGIFY ) { FASTRESTORE_ATOMICOP_WEIGHT = deterministicRandom()->random01() * 200 + 1; }
+	init( BACKUP_ENCRYPTED_SNAPSHOT_USE_TENANT_CACHE, true );
```

Comment:
I'm not sure if you should make this as a configurable parameter? Especially because for backup turning it off won't be feasible for any runs..

Reply:
So currently in production we can have the following scenarios (all of these will have the above knob as true by default):

1. The tenant mode = Disabled on the database so with our check below we will avoid using the tenant cache as it is unnecessary
2. tenant mode = Required so in this case we will always go to the tenant cache, and since the mode is required, cache misses should be infrequent
3. tenant mode = Optional so in this case we will still always go to the tenant cache but if lots of data do not belong to any tenants, performance will be poor

So in production this knob is not very useful. However in simulation we have most of our backup workloads that can run in tenant mode optional but where most of the keys do not belong to any tenants. This causes the workload to perform very poorly and time out. So this is where we use the knob to explicitly turn off the use of the tenant cache. This way we can run our existing backup workloads with our encrypted range file writer but just not go to the cache. We can't really set the tenant mode as disabled for all of these workloads since alot of these workloads test backing up tenant specific system keys

Reply:
I understand the motivation of the change and as I expected the KNOB is make one/few simulation tests not timeout due to the unnecessary lookup. 

IMO, there are multiple ways to solve that problem other than knob

1. 
`if (g_network->isSimulated() && mode == optional) {
   // with approach comment explaining the motivation
   useTenantCache = false; 
}`

2. Inferring the tenant is an `invalid` tenant by keeping tap of the latest known tenant served by the cluster, given the `tenantId` are incremental in nature, it is possible to safely ignore cache lookups. However, the risk of cache not up-to-date (triggers can be falling behind), this option has its gotchas.

I believe #1 option is better than adding a KNOB to support just simulation testing, as you mentioned in production with a TenantMode::Optional the performance would be degraded if most of the tenants are `invalid`, I think have a tighter check `maxValidTenantId` would be helpful; but, for now TenantMode::Optional usecase may not be super interesting to optimize and could be taken up in future adding sufficient TODOs in the code. 
Thoughts?

Reply:
Sure i can implement (1)